src/execution_plan/execution_plan.h

```diff
@@ -88,13 +88,16 @@ OpBase *ExecutionPlan_BuildOpsFromPath(ExecutionPlan *plan, const char **vars,
 /* execution_plan.c */
 
 /* Creates a new execution plan from AST */
-ExecutionPlan *NewExecutionPlan(ResultSet *result_set);
+ExecutionPlan *NewExecutionPlan();
+
+/* Prepare an execution plan for execution: optimize, initialize result set schema. */
+void ExecutionPlan_PreparePlan(ExecutionPlan *plan);
```

Comment:
I think a name like `ExecutionPlan_FinalizePlan` might be better in order to more clearly convey that we're between building and executing the plan.

Reply:
IMO, Finalize has different semantics, such as resource and memory free.